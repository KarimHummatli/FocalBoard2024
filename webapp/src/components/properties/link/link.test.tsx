// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react'
import {render} from '@testing-library/react'

import '@testing-library/jest-dom'
import {IntlProvider} from 'react-intl'

import Link from './link'

const wrapIntl = (children: any) => <IntlProvider locale='en'>{children}</IntlProvider>

describe('components/properties/link', () => {
    test('returns link properties correctly', () => {
        const component = wrapIntl(
            <Link
                value={'https://github.com/mattermost/focalboard'}
                onChange={jest.fn()}
                onSave={jest.fn()}
                onCancel={jest.fn()}
                validator={jest.fn(() => {
                    return true
                })}
            />,
        )

        const {container} = render(component)
        expect(container).toMatchSnapshot()
    })
})
