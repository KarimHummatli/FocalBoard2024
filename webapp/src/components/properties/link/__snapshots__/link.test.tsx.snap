// Jest Snapshot v1, https://goo.gl/fbAQLP


exports[`components/properties/link returns Link correctly 1`] = `
<div>
  <div
    class="URLProperty property-link url"
  >
    <input
      class="Editable octo-propertyvalue"
      placeholder="Empty"
      title="https://github.com/mattermost/focalboard"
      value="https://github.com/mattermost/focalboard"
    />
    <a
      href="https://github.com/mattermost/focalboard"
      rel="noreferrer"
      target="_blank"
    >
      <i
        class="CompassIcon icon-link-variant LinkIcon"
      />
    </a>
  </div>
</div>
`;
